src/Void/Messaging.fs

```diff
@@ -21,10 +21,12 @@ type Channel<'TIn when 'TIn :> Message>
 
     interface Channel with
         member x.publish (message : Message) =
-            match message with
-            | :? 'TIn as msg ->
-                Seq.map (fun handle -> handle msg) _handlers
-            | _ -> Seq.empty
+            if message <> noMessage

```

Comment:
Or what if we didn't ever pass `noMessage` back to the bus at all? i.e. (warning: untested code):

```
member x.publish (message : Message) =
    match message with
    | :? 'TIn as msg ->
        Seq.map (fun handle -> handle msg) _handlers
        |> Seq.filter (fun msg -> msg <> noMessage)
    | _ -> Seq.empty
```

What you've got here is already an improvement, but seems like that would cause the least needless propagation.